### PR TITLE
Convert from KCS (L): Infrastructure Nodes in OpenShift 4

### PR DESCRIPTION
**Pull request by mburke5678**
https://issues.redhat.com/browse/OSDOCS-2077

Bring in information from [Infrastructure Nodes in OpenShift 4](https://access.redhat.com/solutions/5034771) solution

Previews
Machine Management -> [Creating infrastructure machine sets](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html). Added text to the introduction.
[Moving the router](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-router_creating-infrastructure-machinesets). Added code to code block.
[Moving the default registry](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-registry_creating-infrastructure-machinesets). Added code to code block.
[Moving the monitoring solution](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-monitoring_creating-infrastructure-machinesets) Added code to code block.
[Moving OpenShift Logging resources](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-logging_creating-infrastructure-machinesets) Added code to code block.

Nodes -> Nodes -> [Creating infrastructure nodes](https://51466--docspreview.netlify.app/openshift-enterprise/latest/nodes/nodes/nodes-nodes-creating-infrastructure-nodes.html). New assembly with same intro as machine mgmt, procedure to create infra node, link to moving components in machine mgmt.

### CODE REVIEW

File: machine_management/creating-infrastructure-machinesets.adoc [hunk 1]
```diff
@@ -11,10 +11,16 @@ include::modules/machine-user-provisioned-limitations.adoc[leveloffset=+1]
 
 You can use infrastructure machine sets to create machines that host only infrastructure components, such as the default router, the integrated container image registry, and the components for cluster metrics and monitoring. These infrastructure machines are not counted toward the total number of subscriptions that are required to run the environment.
 
+In a production deployment, it is recommended that you deploy at least three machine sets to hold infrastructure components. Both OpenShift Logging and {SMProductName} deploy Elasticsearch, which requires three instances to be installed on different nodes. Each of these nodes can be deployed to different availability zones for high availability. A configuration like this requires three different machine sets, one for each availability zone. In global Azure regions that do not have multiple availability zones, you can use availability sets to ensure high availability.
+
```
Comment (mburke5678) [inline machine_management/creating-infrastructure-machinesets.adoc:15]: ^^ Copied from [Post-install module](https://docs.openshift.com/container-platform/4.11/post_installation_configuration/cluster-tasks.html#post-install-creating-infrastructure-machinesets-production).

**Comment by mburke5678** [comment]
@lyman9966 You have reviewed changes I have made to this assembly. Can you review this PR? Thank you in advance.

**Comment by lyman9966** [comment]
checked Nodes -> Nodes -> [Creating infrastructure nodes](https://51466--docspreview.netlify.app/openshift-enterprise/latest/nodes/nodes/nodes-nodes-creating-infrastructure-nodes.html), it looks good.

**Comment by lyman9966** [comment]
@sunzhaohua2  can you help to review the part of Machine Management -> [Creating infrastructure machine sets](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html) ? 
@lihongan can you help to review the part of [Moving the router](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-router_creating-infrastructure-machinesets) ? 
@juzhao  can you help to review the part of [Moving the monitoring solution](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-monitoring_creating-infrastructure-machinesets) ? 
@xiuwang can you help to review the part of [Moving the default registry](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-registry_creating-infrastructure-machinesets)? 
@QiaolingTang can you help to review the part of [Moving OpenShift Logging resources](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-logging_creating-infrastructure-machinesets)?

**Comment by QiaolingTang** [comment]
Changes in [Moving OpenShift Logging resources](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-logging_creating-infrastructure-machinesets) are LGTM.

**Comment by sunzhaohua2** [comment]
LGTM for [Creating infrastructure machine sets](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html) part

**Comment by xiuwang** [comment]
LGTM for [Moving the default registry](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-registry_creating-infrastructure-machinesets) part

File: modules/infrastructure-moving-monitoring.adoc [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
 
 The monitoring stack includes multiple components, including Prometheus, Thanos Querier, and Alertmanager.
-The Cluster Monitoring Operator manages this stack.
-To redeploy the monitoring stack to infrastructure nodes, you can create and apply a custom config map.
+The Cluster Monitoring Operator manages this stack. To redeploy the monitoring stack to infrastructure nodes, you can create and apply a custom config map.
  
 .Procedure
 
-. Save the following `ConfigMap` definition as the `cluster-monitoring-configmap.yaml` file:
+. Edit the `ingresscontroller` resource and change the `nodeSelector` to use the `infra` label.
```
Comment (juzhao) [inline modules/infrastructure-moving-monitoring.adoc:14]: `Edit the `ingresscontroller` resource and change the `nodeSelector` to use the `infra` label.`
the change is wrong, change to below is better
`Edit the `cluster-monitoring-config` configMap and change the `nodeSelector` to use the `infra` label`

File: modules/infrastructure-moving-monitoring.adoc [hunk 3]
```diff
@@ -24,39 +28,87 @@ metadata:
...
       nodeSelector:
         node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: node-role.kubernetes.io/infra
+        value: reserved
+        effect: NoSchedule
+      - key: node-role.kubernetes.io/infra
+        value: reserved
+        effect: NoExecute
     k8sPrometheusAdapter:
```
Comment (juzhao) [inline modules/infrastructure-moving-monitoring.adoc:60]: missed `grafana` component for 4.6 - 4.10, see from:
https://docs.openshift.com/container-platform/4.10/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-monitoring_creating-infrastructure-machinesets

from 4.11, grafana is removed from stack, no configuration for it is fine
https://docs.openshift.com/container-platform/4.11/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-monitoring_creating-infrastructure-machinesets

**Comment by kalexand-rh** [comment]
Only versions 4.8+ are still in maintenance. I am removing labels from earlier versions from this PR.

File: modules/infrastructure-moving-monitoring.adoc (hunk 3 above)
Comment (mburke5678) [reply modules/infrastructure-moving-monitoring.adoc:60]: I will manually cherrypick these changes and add the `grafana` component there.

**Comment by mburke5678** [comment]
@lihongan can you help to review the part of [Moving the router](https://51466--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-router_creating-infrastructure-machinesets) ?

**Comment by lihongan** [comment]
@mburke5678 the part of "Moving the router" looks good. Thank you!

File: machine_management/creating-infrastructure-machinesets.adoc [hunk 4]
```diff
@@ -100,7 +104,26 @@ include::modules/binding-infra-node-workloads-using-taints-tolerations.adoc[leve
 [id="moving-resources-to-infrastructure-machinesets"]
 == Moving resources to infrastructure machine sets
 
-Some of the infrastructure resources are deployed in your cluster by default. You can move them to the infrastructure machine sets that you created.
+Some of the infrastructure resources are deployed in your cluster by default. You can move them to the infrastructure machine sets that you created by adding the infrastructure node selector.
```
Comment (adellape) [inline machine_management/creating-infrastructure-machinesets.adoc:107]: Colon on end?

**Comment by mburke5678** [comment]
/enterprise-4.11

**Pull request merged**